modules/storage-api/src/main/java/org/apache/ignite/internal/storage/MvPartitionStorage.java

```diff
@@ -30,6 +31,21 @@
  * <p>Each MvPartitionStorage instance represents exactly one partition.
  */
 public interface MvPartitionStorage extends AutoCloseable {
+    /**
+     * Last known replicator index. {@code 0} if index is unknown.
+     */
+    long appliedIndex();
+
+    /**
+     * Sets the last known replicator index.
+     */
+    void appliedIndex(long appliedIndex) throws StorageException;
+
+    /**
+     * {@link #appliedIndex()} value consistent with the data, already persisted on the storage.
+     */
+    long persistedIndex();
```

Comment:
What about the name **persistedAppliedIndex**?

Reply:
That's too much in my opinion

Reply:
Ok